gapic/schema/wrappers.py

```diff
@@ -1067,6 +1078,68 @@ def try_parse_routing_rule(cls, routing_rule: routing_pb2.RoutingRule) -> Option
         params = [RoutingParameter(x.field, x.path_template) for x in params]
         return cls(params)
 
+    @classmethod
+    def resolve(cls, routing_rule: routing_pb2.RoutingRule, request: Union[dict, str]) -> dict:
+        """Resolves the routing header which should be sent along with the request.
+        This function performs dynamic header resolution, identical to what's in `_client_macros.j2`.
+        https://github.com/googleapis/gapic-generator-python/blob/4c5de8791795f8101f6ec66f80b8a8e5e9a21822/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/_client_macros.j2#L150-L164
```

Comment:
nit: Consider using a TinyURL or similar, and enclosing the URL in `<>` so that it's rendered clickable by some tools.

Reply:
Superseded by review comment https://github.com/googleapis/gapic-generator-python/pull/2131#discussion_r1761945891